### PR TITLE
Create setBaseUrl method

### PR DESCRIPTION
**Pull request by ryanannatuvasquez**
## Description
- Create setBaseUrl method to accept url in different environments

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
+## [3.2.1] - 20 May 2019
+### Changed
+- Create setBaseUrl method
+
 ## [3.2.0] - 16 May 2019
 ### Changed
 - remove captcha validation on rejecting jobs
 
-## [3.1.1] - 16 May 2019
+## [3.1.1] - 15 Oct 2018
```
Comment (perseusl) [inline changelog.md]: [question] Why is this changed?

Comment (ryanannatuvasquez) [reply changelog.md]: this is wrong date. if you see in the releases page https://github.com/gengo/gengo-php/releases

File: src/Config.php [hunk 2]
```diff
@@ -183,4 +183,16 @@ public static function setRevisionID($id)
...
+    /**
+     * Set base URL.
+     *
+     * @param string $url URL to be used
+     *
+     * @api
+     */
+    public static function setBaseUrl($url)
+    {
+        self::$_settings['baseurl'] = $url;
```
Comment (shinji-kohara) [inline src/Config.php]: [suggestion] How about check the URL is end with `/` or not and ensure end with it?

Comment (ryanannatuvasquez) [reply src/Config.php]: @shinji-kohara done in https://github.com/gengo/gengo-php/pull/66/commits/dbe46af64cdf32cf8049ceb12609d3d076656a06

PTAL!

File: src/Config.php [hunk 3]
```diff
@@ -183,4 +183,16 @@ public static function setRevisionID($id)
...
+    /**
+     * Set base URL.
+     *
+     * @param string $url URL to be used
+     *
+     * @api
+     */
+    public static function setBaseUrl($url)
+    {
+        self::$_settings['baseurl'] = rtrim($url, '/').'/';;
```
Comment (ryanannatuvasquez) [inline src/Config.php]:
```suggestion
        self::$_settings['baseurl'] = rtrim($url, '/').'/';
```

File: tests/ConfigTest.php [hunk 4]
```diff
@@ -176,4 +176,33 @@ public function testRefusesToAcceptInvalidPreconfiguredRevisionId()
         $this->expectExceptionMessage('Invalid revision ID');
         Config::setRevisionID('wrong_id');
     } //end testRefusesToAcceptInvalidPreconfiguredRevisionId()
+
+
```
Comment (ryanannatuvasquez) [inline tests/ConfigTest.php]:
```suggestion
```

File: tests/ConfigTest.php [hunk 5]
```diff
@@ -176,4 +176,33 @@ public function testRefusesToAcceptInvalidPreconfiguredRevisionId()
...
+        $this->assertEquals($newSettings['baseurl'], $expected);
+    }
+
+    public function setBaseUrlDataProvider()
+    {
+        return [
+            ['https://api.test.gengo.com/', 'https://api.test.gengo.com/'],
+            ['https://api.test.gengo.com', 'https://api.test.gengo.com/'],
+            ['https://api.test.gengo.com//', 'https://api.test.gengo.com/'],
+            ['https://api.test.gengo.com/test/', 'https://api.test.gengo.com/test/'],
```
Comment (perseusl) [inline tests/ConfigTest.php]: [question] Isn't `https://api.test.gengo.com/` is what we want since it is the base url?

**Comment by DarylPenetrante** [comment]
Should we have some kind of validation for acceptable urls?

File: tests/ConfigTest.php (hunk 5 above)
Comment (ryanannatuvasquez) [reply tests/ConfigTest.php]: No. we dont care about if what the client pass here. we dont need to do extra work to fix it. its just changing the base url. to make it work in different environment

**Comment by ryanannatuvasquez** [comment]
@DarylPenetrante Im guessing we dont need to add a validation here. if wrong URL is passed im guessing its the client error not us. we should make this method very simple and straightforward
